src/ucp/api/ucp.h

```diff
@@ -333,6 +335,19 @@ typedef enum {
 
 /**
  * @ingroup UCP_DATATYPE
+ * @brief Generate a reusable identifier for Scatter-gather IOV data type.
```

Comment:
The idea is that lookup will only happen on the first time it is used. If you call send() twice with this "reusable" IOV dtype - first time lookup result will be stored as metadata inside this dtype, and the second call will reuse the previous mapping instead of a second lookup. This means reusable dtypes require allocation, and eventually a call to destroy.
For an IOV - If UMR is enabled, only one (composite) memory key is stored, otherwise all N memory keys are stored. For Stride - if UMR is enabled, only one (strided) memory key is stored, otherwise one contig memory key (for the entire extent) is stored - so that zcopy can happen (if elements are big enough - otherwise bcopy does pack/unpack).